core/services/relay/evm/evm_service.go

```diff
@@ -125,7 +131,94 @@ func (r *Relayer) GetTransactionStatus(ctx context.Context, transactionID common
 		return commontypes.Unknown, err
 	}
 
-	return commontypes.TransactionStatus(status), nil
+	return status, nil
+}
+
+func (r *Relayer) SubmitTransaction(ctx context.Context, txRequest evmtypes.SubmitTransactionRequest) (*evmtypes.TransactionResult, error) {
+	config := r.chain.Config()
+
+	fromAddress := config.EVM().Workflow().FromAddress().Address()
+	var gasLimit uint64
+	if txRequest.GasConfig != nil && txRequest.GasConfig.GasLimit != nil {
+		gasLimit = *txRequest.GasConfig.GasLimit
+	}
+
+	uuid, err := uuid.NewUUID()
+	if err != nil {
+		return nil, err
+	}
+	txID := uuid.String()
+
+	// PLEX-1524 - review which transmitter checker we should use
+	var checker evmtxmgr.TransmitCheckerSpec
+	checker.CheckerType = evmtxmgr.TransmitCheckerTypeSimulate
+	value := big.NewInt(0)
+
+	// PLEX-1524 - Define how we should properly get the workflow execution ID into the meta without making the API CRE specific.
+	var txMeta *txmgrtypes.TxMeta[common.Address, common.Hash]
+	txmReq := evmtxmgr.TxRequest{
+		FromAddress:    fromAddress,
+		ToAddress:      txRequest.To,
+		EncodedPayload: txRequest.Data,
+		FeeLimit:       gasLimit,
+		Meta:           txMeta,
+		IdempotencyKey: &txID,
+		// PLEX-1524 - Review strategy to be used.
+		Strategy: txmgr.NewSendEveryStrategy(),
+		Checker:  checker,
+		Value:    *value,
+	}
+
+	_, err = r.chain.TxManager().CreateTransaction(ctx, txmReq)
+	if err != nil {
+		return nil, fmt.Errorf("%w; failed to create tx", err)
+	}
+
+	maximumWaitTimeForConfirmation := config.EVM().ConfirmationTimeout()
+	start := time.Now()
+	for {
+		txStatus, txStatusErr := r.chain.TxManager().GetTransactionStatus(ctx, txID)
+		if txStatusErr != nil {
+			return nil, txStatusErr
+		}
+		if txStatus == commontypes.Fatal {
+			return &evmtypes.TransactionResult{
+				TxStatus: evm.TxFatal,
+				TxHash:   evmtypes.Hash{},
+			}, nil
+		}
+		if txStatus == commontypes.Unconfirmed || txStatus == commontypes.Finalized {
+			break
+		}
+		if time.Since(start) > maximumWaitTimeForConfirmation {
+			return nil, errors.Errorf("Wait time for Tx %s to get confirmed was greater than maximum wait time %d", txID, maximumWaitTimeForConfirmation)
```

Comment:
IMHO this at least needs to be precise about the tx being sent and _still pending_, and it seems odd to leave them with no recourse in that case - why not just return pending so they have the option of following up?